airbyte-commons-temporal/src/main/java/io/airbyte/commons/temporal/scheduling/ConnectionNotificationWorkflow.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.temporal.scheduling;
+
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+import java.io.IOException;
+import java.util.UUID;
+
+@WorkflowInterface
+public interface ConnectionNotificationWorkflow {
+
+  @WorkflowMethod
+  boolean sendSchemaChangeNotification(UUID connectionId, boolean isBreaking) throws IOException, InterruptedException, ApiException;
```

Comment:
Does it make sense to have `boolean sendSchemaChangeNotification(UUID connectionId)` instead and have the activity call the API to retrieve the details of the connectionId.
It seems to me that if we want to have a better message than just a `connectionId` which is going to be pretty cryptic for users to read, we need to query something to get more details. Passing `isBreaking` here would just be redudant in the end.

Reply:
the thing i was thinking is that the scheduler handler that calls this workflow will already have retrieved the connection from the db so we can avoid doing it twice. I think we probably will want to pass in more data about the connection - haven't gotten full details yet so this was sort of a mockup but probably passing in the name does make sense, and I realized we also have a notifySchemaChanges boolean option that we should pass in as well.

Reply:
Performance wise, I don't expected we'd be hitting a bottleneck here, we are looking at an extra lookup when we detect a schema change.
The benefit I see of not passing the details in the workflow interface is that it would remove the need for some versioning in temporal. (My understanding is that every time we modify some IO, we should have some versioning logic to ensure the temporal replayability constraint still holds)

The part that is not clear in this case is how much do we want the caller to control what's in the actual notification. Currently it looks like the config and formatting is all handled in the activity. It feels easier to have everything in the same place rather than split those operation. That being said, I don't have a clear vision on what we want to do with notifications in the future, if you think it makes more sense, this works for me.

Reply:
I think this makes sense - I wasn't thinking about temporal replayability. I'll update this so it just takes a connectionId and then the notifySchemaChange activity can fetch the connection details

Reply:
started implementing this, but also thinking about it more - we may want to include diff info in the message (not sure), and if so, we calculate the catalog diff in the scheduler handler endpoint, and would probably want to pass it into the workflow instead of recalculating the diff as well, since that isn't stored anywhere. what do you think?

Reply:
I am thinking about the notifications itself, we can add more details. One note here is that eventually we still need a user to go and fix the configuration so as long as we can highlight the class of issue (breaking change for example), I think it's a good start.
This feels like something that we may want to reconsider in the future, for now, I'd punt this to when we have more feedback on notifications. This should remain an easy API change if we want to do it.